File: ios/GitaVani/GitaVani/Views/Common/ResumeReadingBanner.swift
import SwiftUI

struct ResumeReadingBanner: View {
    let chapter: Int
    let verse: Int
    let chapterName: String
    let theme: AppTheme
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 12) {
                Image(systemName: "book.fill")
                    .font(.title3)
                    .foregroundStyle(theme.accentColor)

                VStack(alignment: .leading, spacing: 2) {
                    Text("Continue Reading")
                        .font(.subheadline)
                        .fontWeight(.semibold)
                        .foregroundStyle(theme.primaryTextColor)
                    Text("Chapter \(chapter): \(chapterName) — Verse \(verse)")
                        .font(.caption)
                        .foregroundStyle(theme.secondaryTextColor)
                }

                Spacer()

                Image(systemName: "chevron.right")
                    .font(.caption)
                    .foregroundStyle(theme.secondaryTextColor)
            }
            .padding()
            .background(theme.cardBackgroundColor)
            .cornerRadius(12)
        }
        .buttonStyle(.plain)
    }
}

File: android/GitaVani/app/src/main/java/com/nikhilsi/gitavani/ui/common/ResumeReadingBanner.kt
package com.nikhilsi.gitavani.ui.common

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.MenuBook
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.nikhilsi.gitavani.theme.AppTheme

@Composable
fun ResumeReadingBanner(
    chapter: Int,
    verse: Int,
    chapterName: String,
    theme: AppTheme,
    onTap: () -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(theme.cardBackgroundColor)
            .clickable(onClick = onTap)
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Icon(
            imageVector = Icons.AutoMirrored.Filled.MenuBook,
            contentDescription = "Resume reading",
            tint = theme.accentColor
        )

        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = "Continue Reading",
                fontSize = 14.sp,
                fontWeight = FontWeight.SemiBold,
                color = theme.primaryTextColor
            )
            Text(
                text = "Chapter $chapter: $chapterName — Verse $verse",
                fontSize = 12.sp,
                color = theme.secondaryTextColor
            )
        }

        Icon(
            imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
            contentDescription = null,
            tint = theme.secondaryTextColor
        )
    }
}
